src/main.cpp

```diff
@@ -3977,7 +3977,7 @@ bool CBlock::CheckBlock(std::string sCaller, int height1, int64_t Mint, bool fCh
 
         // Verify beacon contract if a transaction contains a beacon contract
         if (!VerifyBeaconContractTx(tx.hashBoinc) && !fLoadingIndex)
-            return DoS(25, error("CheckBlock[] : bad beacon contract found in tx contained within block; rejected"));
+            return DoS(25, error("CheckBlock[] : bad beacon contract found in tx %s contained within block; rejected", tx.GetHash().ToString().c_str()));
```

Comment:
You can skip your beaconchecks here.

Reply:
? why. this is is a malicious beacon contract gets in the block manually by attacker this prevent it from being accepted in chain. because only the staker would have the bad contract if all other clients prevent it from memory pool

Reply:
I was talking about skipping the checks here only if the block is below mandatory.
I done it already in my pr #700 .

Reply:
ahh, sorry for the confusion :)

Reply:
I would appreciate your review on that PR. If it is correct, ...